src/main/java/uk/ac/ebi/intact/app/internal/ui/components/query/AdvancedSearchQueryComponent.java

```diff
@@ -0,0 +1,261 @@
+
+        queryTextField.setText(getQueriesFromRuleBuilders(rules, queryOperators.getRuleOperator()));
+        rulesPanel.add(rulePanel);
+        rulesPanel.revalidate();
+        rulesPanel.repaint();
+    }
+
+    private JButton getDeleteRuleButton(JPanel rulePanel) {
+        JButton deleteRuleButton = new JButton("Delete rule");
+        setButtonIntactPurple(deleteRuleButton);
+        deleteRuleButton.addActionListener(e -> {
+            deleteRule(rulePanel);
+        });
+        return deleteRuleButton;
+    }
+
+    public void deleteRule(JPanel oneRuleBuilder) {
+        rulesPanel.remove(oneRuleBuilder);
+
+        queryTextField.setText(getQueriesFromRuleBuilders(rules, queryOperators.getRuleOperator()));
+        rulesPanel.revalidate();
+        rulesPanel.repaint();
+    }
+
+    private JPanel getQueryInputField(){
+        JPanel queryInputFieldContainer = new JPanel();
+
+        queryTextField.setMinimumSize(new Dimension(frameWidth, 25));
+        queryTextField.setPreferredSize(new Dimension(frameWidth/2, 25));
+        queryTextField.setVisible(true);
+
+        queryInputFieldContainer.add(queryTextField);
+
+        return queryInputFieldContainer;
+    }
+
+    public String getQueryFromTextField(){
+        return queryTextField.getText();
+    }
+
+    private void modifyComboBoxesFromTextField(String queryFromTextField){
+        //todo: remove every combo boxes and build from scratch
+
+        String[] queries = getQueriesFromTextField(queryFromTextField);
+        modifyQueryBuildersNumberFromTextField(queries);
+
+        for (int i = 0; i < queries.length; i++) {
+            for (Field field : Field.FIELD_MAP.values()) {
+                if (queryFromTextField.contains(field.getMiqlQuery())) {
+                    QueryComponents queryComponents = parseQuery(queries[i].trim());
+                }
+            }
+        }
+    }
+
+    private void modifyQueryBuildersNumberFromTextField(String[] queries){
+        while (queries.length > rules.size()) {
+            addRule();
+        }
+        while (queries.length < rules.size()) {
+            Component lastRule = rulesPanel.getComponents()[rules.size() - 1];
+            deleteRule((JPanel) lastRule);
+        }
+
+        rulesPanel.revalidate();
+        rulesPanel.repaint();
+        queryTextField.setText(getQueriesFromRuleBuilders(rules, queryOperators.getRuleOperator()));
+    }
+
+    private String[] getQueriesFromTextField(String queryFromTextField){
+        String[] queries;
+
+        if (queryFromTextField.contains("AND")) {
+            queries = queryFromTextField.split("AND");
+        } else if (queryFromTextField.contains("OR")) {
+            queries = queryFromTextField.split("OR");
+        } else {
+            queries = new String[]{queryFromTextField};
+        }
+
+        return queries;
+    }
+
+    private String getOperatorFromTextField(String queryFromTextField){
+        if (queryFromTextField.contains("NOT")){
```

Comment:
Here we're checking `"NOT"`. In line 213 we're checking `"NOT "`. We should be consistent.

Reply:
Not that important, just a small detail.